mixins/interactive-mixin.js

```diff
@@ -0,0 +1,115 @@
+import { css, html } from 'lit';
+import { findComposedAncestor, isComposedAncestor } from '../helpers/dom.js';
+import { forceFocusVisible, getNextFocusable } from '../helpers/focus.js';
+import { ifDefined } from 'lit/directives/if-defined.js';
+import { LocalizeCoreElement } from '../helpers/localize-core-element.js';
+
+const keyCodes = {
+	ENTER: 13,
+	ESCAPE: 27
+};
+
+export const InteractiveMixin = superclass => class extends LocalizeCoreElement(superclass) {
+
+	static get properties() {
+		return {
+			_hasInteractiveAncestor: { state: true },
+			_interactive: { state: true }
+		};
+	}
+
+	static get styles() {
+		return css`
+			.interactive-container.focus-visible,
+			.interactive-container:focus-visible {
+				border-radius: 6px;
+				outline: 2px solid var(--d2l-color-celestine);
+				outline-offset: 2px;
+			}
+		`;
+	}
+
+	constructor() {
+		super();
+		this._hasInteractiveAncestor = false;
+		this._interactive = false;
+	}
+
+	connectedCallback() {
+		super.connectedCallback();
+
+		const parentGrid = findComposedAncestor(this.parentNode, node => {
+			return ((node.tagName === 'D2L-LIST' && node.grid) || (node.nodeType === Node.ELEMENT_NODE && node.getAttribute('role') === 'grid'));
+		});
+		this._hasInteractiveAncestor = (parentGrid !== null);
+	}
+
+	focus() {
+		if (!this.shadowRoot) return;
+		if (this._hasInteractiveAncestor && !this._interactive) forceFocusVisible(this.shadowRoot.querySelector('.interactive-container'));
+		else this._focusInteractiveContent();
```

Comment:
Having them implement this method makes the contract a little weird... how awkward would it be if they passed this in alongside the label as a delegate function to call?

Reply:
Yep, that would be fine too.  👍